src/types/local_dependency_config.go

```diff
@@ -1,8 +0,0 @@
-package types
-
-// LocalDependencyConfig represents the configuration of a local dependency
-type LocalDependencyConfig struct {
-	Persist               []string          `yaml:",omitempty"`
-	DependencyEnvironment map[string]string `yaml:"dependency-environment,omitempty"`
-	ServiceEnvironment    map[string]string `yaml:"service-environment,omitempty"`
```

Comment:
De-nested these from under Config with the given changes

* Renamed `DependencyEnvironment` to `EnvironmentVariables` for passing only to the dependency
* Added `Secrets` to be consistent with other passing of environment variables
* Removed `ServiceEnvironment` as we can use app wide environment variables to achieve the same thing